### PR TITLE
NewHaven TFT display driver

### PR DESCRIPTION
**Pull request by Jake-Carter**
Migrated from #9 (No idea why, but Github automatically closed it after I pushed my changes to Victor's fork...)

Only changes I made between #9 and here are:
* https://github.com/Analog-Devices-MSDK/Libraries-miscDrivers/commit/3c681c04c6b01804c17c6bfb22d1733dac7b9e76 (add mono display function)
* https://github.com/Analog-Devices-MSDK/Libraries-miscDrivers/commit/092b84df4577a2dc13a04858fa7872f4ae498352 (clang-format)
* https://github.com/Analog-Devices-MSDK/Libraries-miscDrivers/commit/876ec38b1d1af60c5f864540b0f50f4df8504081 (lint)

Drivers work great.  TFT Demo passes, and I tested mono and color display functions with CameraIF below:

(`CAMERA=HM0360_MONO`)

(`CAMERA=OV7692`)

### CODE REVIEW

File: Display/tft_st7789v.c [hunk 1]
```diff
@@ -0,0 +1,870 @@
...
+    MXC_TFT_Backlight(1);
+
+    return E_NO_ERROR;
+}
+
+void MXC_TFT_SetRotation(tft_rotation_t rotation)
+{
+    tft_rotation = rotation;
+
+    __disable_irq();
```
Comment (lorne-maxim) [inline Display/tft_st7789v.c:479]: Need to save state of IRQ before disabling it, then only enable it at the end if it was already on.  (This is a common error throughout the SDK that we need to get cleaned up.)

File: Display/tft_st7789v.h [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+ * @ingroup bsp
+ * @defgroup tft_st7789v NewHaven TFT
+ * @{
+ */
+
+#define DISPLAY_WIDTH 320 /**< TFT width in pixels. */
+#define DISPLAY_HEIGHT 240 /**< TFT height in pixels. */
+#define TFT_SPI_FREQ 20000000 /**< SPI clock frequency in Hertz. */
+
+#if (TARGET_NUM == 78000)
```
Comment (lorne-maxim) [inline Display/tft_st7789v.h:58]: These are board specific options and should be configured in the board files.  Let's keep the miscDrivers as board agnostic as possible.

**Review by lorne-maxim — CHANGES_REQUESTED** [review]
Most the changes requested revolve around making these file less board specific.  Ideally, we can move all board specific settings to board.h (or maybe board.mk, but that feels a little less intuitive to me).

File: Display/tft_st7789v.c [hunk 3]
```diff
@@ -0,0 +1,870 @@
...
+    while (!(spi->intfl & MXC_F_SPI_INTFL_MST_DONE)) {}
+
+    spi->intfl = spi->intfl;
+}
+
+static void spi_transmit_data_buf(void *src, int count)
+{
+    uint8_t *buf = (uint8_t *)src;
+
+    MXC_GPIO_OutClr(TFT_SS_PORT, TFT_SS_PIN);
```
[the remark this answers is not in the document]
Comment (Jake-Carter) [reply Display/tft_st7789v.c]: @lorne-maxim done, SPI transport layer has been moved to the board files.

File: Display/tft_st7789v.c (hunk 1 above)
Comment (Jake-Carter) [reply Display/tft_st7789v.c:375]: Done, see https://github.com/Analog-Devices-MSDK/Libraries-miscDrivers/pull/18/commits/e804a9bc41270ef5bf9f2c977291a00c998dad3b

File: Display/tft_st7789v.c [hunk 4]
```diff
@@ -0,0 +1,870 @@
...
+
+void MXC_TFT_SetFont(int font_id)
+{
+    if (font_id != 0)
+        set_font((unsigned char *)font_id);
+}
+
+void MXC_TFT_Printf(const char *format, ...)
+{
+    (void)format;
```
[the remark this answers is not in the document]
Comment (Jake-Carter) [reply Display/tft_st7789v.c:708]: Yes, but it is only used by 4 examples across our whole SDK.  Additionally, it isn't used by the AI87 TFT_Demo example.

The only file that implements it is the ssd2119 file, and the implementation is not very clean...

https://github.com/Analog-Devices-MSDK/Libraries-miscDrivers/blob/726518622a423cd8fa6a5a7168ced57364dc680a/Display/tft_ssd2119.c#L1286

I think we should address this in a future PR alongside some other TFT standardizations.

File: Display/tft_st7789v.c [hunk 5]
```diff
@@ -0,0 +1,870 @@
...
+}
+
+void MXC_TFT_ConfigPrintf(area_t *area)
+{
```
[the remark this answers is not in the document]
Comment (Jake-Carter) [reply Display/tft_st7789v.c:712]: Same here, not currently required for the AI87 EVKIT but should be addressed in a future PR

File: Display/tft_st7789v.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (lorne-maxim) [reply Display/tft_st7789v.c:708]: Sounds good.  Please create a Jira or GitHub issues with the list of improvements you have in mind.

**Review by lorne-maxim — APPROVED** [review]

**Pull request merged**